src/Service/BesluitToXxllncService.php

```diff
@@ -0,0 +1,400 @@
+
+        $logMessage = "Posting relation for case (besluit): $besluitCaseSourceId to normal case: $caseSourceId";
+        $endpoint   = "/case/$besluitCaseSourceId/relation/add";
+        $bodyString = json_encode(['related_id' => $caseSourceId]);
+
+        // Send the POST/PUT request to xxllnc.
+        try {
+            isset($this->style) === true && $this->style->info($logMessage);
+            $response = $this->callService->call($xxllncAPI, $endpoint, 'POST', ['body' => $bodyString]);
+            $result   = $this->callService->decodeResponse($xxllncAPI, $response);
+
+            $caseId = $result['result']['reference'] ?? null;
+        } catch (Exception $e) {
+            isset($this->style) === true && $this->style->error("Failed to set relation for besluit case to normal case, message:  {$e->getMessage()}");
+
+            return false;
+        }//end try
+
+        return true;
+
+    }//end sendCaseRelationForBesluit()
+
+
+    /**
+     * Maps zgw besluit to xxllnc case.
+     *
+     * @param string       $besluittypeSourceId The besluittype source id.
+     * @param ObjectEntity $besluitObject       ZGW Besluit object.
+     * @param ObjectEntity $zaakObject          Zgw Zaak object.
+     *
+     * @throws Exception
+     *
+     * @return string|null $this->data Data which we entered the function with.
+     */
+    public function mapBesluitToXxllnc(string $besluittypeSourceId, ObjectEntity $besluitObject, ObjectEntity $zaakObject): ?string
+    {
+        $this->zgwToXxllncService->xxllncZaakSchema = $this->resourceService->getSchema('https://development.zaaksysteem.nl/schema/xxllnc.zaakPost.schema.json',                  'common-gateway/xxllnc-zgw-bundle');
+        $this->zgwToXxllncService->xxllncAPI        = $this->resourceService->getSource('https://development.zaaksysteem.nl/source/xxllnc.zaaksysteem.source.json',               'common-gateway/xxllnc-zgw-bundle');
+        $xxllncZaakMapping                          = $this->resourceService->getMapping('https://development.zaaksysteem.nl/mapping/xxllnc.ZgwBesluitToXxllncCase.mapping.json', 'common-gateway/xxllnc-zgw-bundle');
+
+        $zaakArrayObject = $zaakObject->toArray();
+        $bsn             = ($zaakArrayObject['rollen'][0]['betrokkeneIdentificatie']['inpBsn'] ?? $zaakArrayObject['verantwoordelijkeOrganisatie'] ) ?? null;
+        if ($bsn === null) {
+            $this->logger->error('No bsn found in a rol->betrokkeneIdentificatie->inpBsn or verantwoordelijkeOrganisatie.');
+
+            return null;
+        }
+
+        $besluitObjectArray  = array_merge($besluitObject->toArray(), ['bsn' => $bsn, 'caseTypeId' => $besluittypeSourceId]);
+        $besluitMappingArray = $this->mappingService->mapping($xxllncZaakMapping, $besluitObjectArray);
+        unset($besluitMappingArray['zgwZaak']);
```

Comment:
Maybe you can unset the zgwZaak in the mapping

Reply:
Removed because not needed. This happens in a unset mapping